src/Evolution/TypeTraits.hpp

```diff
@@ -23,4 +25,24 @@ template <class T>
 constexpr bool has_analytic_solution_alias_v =
     has_analytic_solution_alias<T>::value;
 // @}
+
+/// \ingroup AnalyticDataGroup
+template <typename T>
+using is_analytic_data =
+    typename std::is_convertible<T*, detail::MarkAsAnalyticData*>;
```

Comment:
yes, `is_base_of` will match both public and private base classes, while `is_convertible` will only work for public base classes.  `MagnetizedFmDisk` inherits privately from `FishboneMoncriefDisk` to use its implementation.   And it would be a lot messier to contain an object rather than use private inheritance